### ISSUE TITLE
[BUG] History api with agents

### ISSUE THREAD

**Issue by takefy-dev** [FlowiseAI/Flowise#3287]
**Describe the bug**
With the agents and prediction api I want to avoid using agent memory node since it's sqlite and handle my own memory system so i fault using the history  key in the api, but it somehow does not work even with the agent memory node. 

**To Reproduce**

1. Go to 'Agent'
2. Use the api

**Expected behavior**
Using the history 

**Screenshots**
Exemple: 
```
{
    "chatId": "441217426-qqq",
    "history": [
        {
            "role": "userMessage",
            "content": "\u003cp\u003ecoucou ca va\u003c/p\u003e"
        },
        {
            "role": "apiMessage",
            "content": "coucouuu 😘 ça va bien et toi ? qu’est-ce qui t’amène ici aujourd'hui ? 😏 dis-moi un peu sur toi, tu t'appelles comment ?"
        },
         {
            "role": "userMessage",
            "content": "je m'appelle jules"
        },
        {
            "role": "apiMessage",
            "content": "salut jules 😘, enchantée de te rencontrer ! dis-moi, tu fais quoi dans la vie ? et t'es d'où ? 🌍✨"
        }
    ],
    "question": "je m'appelle comment?"
}

```
but it output    "text": "coucou toi! je sais pas encore... tu vas me dire ton petit nom ?? 😘 et peut-être aussi ton âge, j'aime bien savoir à qui je parle 😏",

which means the llm does not know my name since it's is well present in the history in the api call.

**Setup**

-   Installation [e.g. docker]
-   Flowise Version [2.1.1]
-   OS: [Linux]
-   Browser [chrome]

**Comment by HenryHengZJ** [comment]
You need to have an Agent Memory attached in order for history to work:
![image](https://github.com/user-attachments/assets/c42da825-0d5c-40be-b1a0-0a88bd0a6507)

![image](https://github.com/user-attachments/assets/fb727994-d584-4acd-94e6-23b39f096c7b)

**Comment by takefy-dev** [comment]
yeah but i when to handle my own history and no it does not work if a chat id is already started it does not overwirte the current history...

**Comment by HenryHengZJ** [comment]
we will add more support of Agent Memory to MySQL and PostgreSQL

**Comment by takefy-dev** [comment]
> we will add more support of Agent Memory to MySQL and PostgreSQL
hey thx but when ?? been 2 weeks still no support running on a uge ecommerce website already got 2k2 session in 4 days, sqlite is slow it makes crash flowise
